TOOLS/lua/autocrop.lua

```diff
@@ -18,97 +18,225 @@
 -- is unambiguous (i.e., not a black frame, black background title
 -- card, or dark scene).
 --
--- The default delay between insertion of the cropdetect and
--- crop filters may be overridden by adding
---
--- --script-opts=autocrop.detect_seconds=<number of seconds>
---
--- to mpv's arguments. This may be desirable to allow cropdetect more
--- time to collect data.
+-- The default options can be overridden by adding
+-- --script-opts-append=autocrop-<parameter>=<value>
+-- into mpv.conf
+
+
+
 require "mp.msg"
+require 'mp.options'
 
-script_name = mp.get_script_name()
-cropdetect_label = string.format("%s-cropdetect", script_name)
-crop_label = string.format("%s-crop", script_name)
 
--- number of seconds to gather cropdetect data
-detect_seconds = tonumber(mp.get_opt(string.format("%s.detect_seconds", script_name)))
-if not detect_seconds then
+-- auto: bool - Whether to automatically apply crop at the start of playback.
+-- auto_keep: bool - If crop already exists, decision whether to keep it.
+-- detect_limit: number[0-255] - Black threshold for cropdetect. See limit of https://ffmpeg.org/ffmpeg-filters.html#cropdetect
+-- detect_round: number[2^n] - 2 has best accuracy, 16 has best compatibility. See round of https://ffmpeg.org/ffmpeg-filters.html#cropdetect
+-- detect_min_ratio: number[0.0-1.0] - The ratio of the minimum clip size to the original.
+-- detect_seconds: seconds - How long to gather cropdetect data.
+local options = {
+    auto = true,
+    auto_keep = false,
+    detect_limit = 16,
+    detect_round = 4,
+    detect_min_ratio = 0.8,
```

Comment:
This is actually too conservative. For example, to play an old 4:3 DVD with a letterboxed 2.4:1 film, The cropped height will be .557. Even a 16:9 image inscribed onto 4:3 will be 0.75. I think 0.5 would probably be a more generic default.